packages/client/src/runtime/core/init/checkPlatformCaching.test.ts

```diff
@@ -1,31 +1,87 @@
+import { PrismaClientInitializationError } from '@prisma/engine-core'
+
 import { checkPlatformCaching } from './checkPlatformCaching'
 
 test('generated via postinstall on vercel', () => {
+  const consoleMock = jest.spyOn(global.console, 'error').mockImplementation()
+
   expect(() => checkPlatformCaching({ postinstall: true, ciName: 'Vercel', clientVersion: '0.0.0' }))
     .toThrowErrorMatchingInlineSnapshot(`
-    We have detected that you've built your project on Vercel, which caches dependencies.
-    This leads to an outdated Prisma Client because Prisma's auto-generation isn't triggered.
-    To fix this, make sure to run the \`prisma generate\` command during your build process.
+    Prisma has detected that this project was built on Vercel, which caches dependencies. This leads to an outdated Prisma Client because Prisma's auto-generation isn't triggered. To fix this, make sure to run the \`prisma generate\` command during the build process.
+
     Learn how: https://pris.ly/d/vercel-build
   `)
+
+  expect(consoleMock.mock.calls[0]).toMatchInlineSnapshot(`
+    [
+      Prisma has detected that this project was built on Vercel, which caches dependencies. This leads to an outdated Prisma Client because Prisma's auto-generation isn't triggered. To fix this, make sure to run the \`prisma generate\` command during the build process.
+
+    Learn how: https://pris.ly/d/vercel-build,
+    ]
+  `)
 })
 
 test('generated on vercel', () => {
+  const consoleMock = jest.spyOn(console, 'error').mockImplementation()
+
   expect(() => checkPlatformCaching({ postinstall: false, ciName: 'Vercel', clientVersion: '0.0.0' })).not.toThrow()
+
+  expect(consoleMock.mock.calls[0]).toMatchInlineSnapshot(`
```

Comment:
Why do we print an error in this case?

Reply:
It ensures that the _error_ is shown in a readable format in Netlify and Vercel logs UI instead of being hidden in the general "crash" output or whatever the user has configured their app to do.

Reply:
Yes, but my question is about this test where `postinstall` is `false` and we check that no error is thrown.

Reply:
Ah, this makes sense now. Thank you!